docs/source/how-to/migrations/upgrade-to-16.md

```diff
@@ -0,0 +1,70 @@
+# Upgrading your OAuthenticator to version 16.0
+
+The following section describes what to keep in mind when upgrading to OAuthenticator 16.0.
+
+````{important}
+(token_request)=
+
+```{rubric} The token request
+```
+Whenever _token request_ is used, it refers to the HTTP request exchanging the [oauth code](https://www.rfc-editor.org/rfc/rfc6749#section-1.3.1) for the [access token](https://www.rfc-editor.org/rfc/rfc6749#section-1.4).
+
+This request is sent to the [`token_url`](https://oauthenticator.readthedocs.io/en/latest/reference/api/gen/oauthenticator.oauth2.html#oauthenticator.oauth2.OAuthenticator.token_url) in [`get_token_info()`](https://oauthenticator.readthedocs.io/en/latest/reference/api/gen/oauthenticator.oauth2.html#oauthenticator.oauth2.OAuthenticator.get_token_info) method.
```

Comment:
This is a document parsed by sphinx, so you can make reference that doesn't write like `https://oauthenticator` etc, but instead use the sphinx relative references.

Even the generated reference fields will be sphinx linkable I'm quite sure, hmmm, but how? I think its done by `autodoc_traits` which relies on `sphinx.ext.autodoc`... Hmmm...

I suspect jupyterhub/jupyterhub has made use of this. Maybe something like... Hmmm...

```
{class}`OAuthenticator`
{class}`oauthenticator.oauth.OAuthenticator`
```

@minrk do you have knowledge on how to link to autodoc_traits generated functions/classes etc?

Reply:
Good idea @consideRatio! I found this reference in the `jupyterhub` docs for example and I'll try to play with it and see how it goes.

https://github.com/jupyterhub/jupyterhub/blob/36f74689c4dd7c9825da1fae9175469dde926cf9/docs/source/reference/authenticators.md?plain=1#L3

Reply:
Ah nice! Yes that is exactly what I was thinking about!

I see that if one clicks it, one goes from...

![image](https://user-images.githubusercontent.com/3837114/213677527-dc641011-fb0b-4e37-9b02-2de2eb1677c9.png)

To...

![image](https://user-images.githubusercontent.com/3837114/213677624-17574883-5029-439b-b38f-e6d9ce56e975.png)